WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -32,7 +32,56 @@ def new
   end
 
   def index
-    @competitions = Competition.all.select([:id, :name, :cityName, :countryId]).order(:year, :month, :day).reverse_order
+    query = "CAST(CONCAT(year,'-',month,'-',day) as Datetime) > ? and showAtAll = true"
+    query_params = [(Date.today - 90)]
+
+    @regions = [ ["All","all"],["",""],["Africa","_Africa"],["Asia","_Asia"],["Europe","_Europe"],["North America","_North America"],["Oceania","_Oceania"],["South America","_South America"],["",""] ] + Country.all.map { |country| [country.name, country.id] }
+    @events = [ ["All", "all"], ["",""] ] + Event.all_official.map { |event| [event.name, event.id] }
+    @years = [ ["Current","current"],["All","all"],["",""] ] + Competition.select(:year).map(&:year).uniq.reverse!
+    @competitions = Competition.where(showAtAll: true).order(:year, :month, :day).reverse_order
+
+    # This need to be the first thing, otherwise @competitions will be an array instead of an object
+    # and the .where will not work
+    if params[:years]
+      if params[:years] == "current"
+        @competitions = @competitions.where(query, query_params)
+      elsif params[:years] != "all"
+        @competitions = @competitions.reject { |competition| competition.year.to_s != params[:years] }
+      end
+    else
+      @competitions = @competitions.where(query, query_params)
```

Comment:
This else seems like it could be consolidated with the `... == "current"` clause above. You did stuff like this lower on, how about something like:

```ruby
if params[:years].blank?
  params[:years] = "current"
end
```